src/Input/AbstractParamAwareInput.php

```diff
@@ -54,6 +55,11 @@ public function __construct(InputInterface $input, OutputInterface $output, Ques
         $this->input  = $input;
         $this->output = $output;
         $this->helper = $helper;
+
+        Assert::allIsInstanceOf($params, InputParamInterface::class, sprintf(
```

Comment:
This is redundant with the declared `@param`

Reply:
Okay, but how do I ensure that this is true once a value is passed to the constructor? Since the value type is `array`, it could be _anything_; this assertion ensures it's what I need it to be.

Reply:
No, it cannot be anything: you declared it to be `@param array<string, InputParamInterface> $params`, and that is to be respected by the caller. It is no longer our responsibility to check it.